### DISCUSSION TITLE
Figure out how to order lists of collections and field names in admin dropdowns - alphabetical or grouped

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#14121] [Feature Requests]
![Screenshot from 2022-06-26 18-51-12](https://user-images.githubusercontent.com/95102939/175827363-9e79f80b-5f5c-4d41-ac5b-e45385f81512.png)
![Screenshot from 2022-06-26 18-51-39](https://user-images.githubusercontent.com/95102939/175827366-8b30a96c-5b28-493f-b3b8-026b79ce5be4.png)

In first image, the order for tables is alphabetical, but in the second image, fields are listed in the order they are in the collection. Should the fields be shown alphabetically as the tables for consistency or in the order the admin has put them? Fields are listed alphabetically in the foreign key input for O2Ms.

Are table names being shown have anything to do with junction tables?

I think the bigger problem that needs to be decided upon: should tables and fields be listed alphabetically, or as the admin has set them?

e.g. if the admin groups collections together in a folder or as sub collections, then should they be shown like this in the above (and other) listings, with a divider to separate the groupings?
i.e.
An admin has a collection of cars, engines, wheels, manufacturers and races: if grouped together by the admin in the model like this:
- cars
  - engines
  - wheels
- manufacturers
- races

Then, the dropdown could show the collections to mirror this:

- cars
- engines
- wheels
- --------
- manufacturers
- races
- --------

No divider between manufacturers and races because they're on the same level.

Personally, I prefer the grouped format over alphabetical because it's based on how I the user have decided that it makes sense to group things, rather than if it were alphanumerical, looking for the races collection between manufacturers and wheels which makes no system/logical sense. This reduces friction between the user, Directus, and the system that's being created.

The same could be done for fields if grouped together in panels, or, in the future, in tab panels.

---

Also related: it just struck me, that in the first image, tableNames are shown, not Collection Names, whilst it's asking for a collection name! (e.g. "fooBars" instead of "Foo Bars") and fields are correctly shown as their display title, not column name in the second image

**Comment by rijkvanzanten** [comment]
> Personally, I prefer the grouped format over alphabetical because it's based on how I the user have decided that it makes sense to group things

Agreed! I believe the fields dropdown already adheres to groups the way you described. Regarding the "name" vs the "key", names should be used for both for consistency's sake 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
